src/dao/AlarmDAO.ts

```diff
@@ -438,6 +441,22 @@ export class AlarmDAO extends AbstractDAO<number, OnmsAlarm> {
    * @hidden
    */
   protected async getOptions(filter?: Filter): Promise<OnmsHTTPOptions> {
+    if (filter) {
+      this.visitFilter(filter, {
+        onRestriction: (restriction: Restriction) => {
+          if (restriction.attribute === 'isAcknowledged') {
+            let value = String(restriction.value).toLowerCase() === 'true';
+            restriction.attribute = 'alarmAckTime';
+            if (restriction.comparator.label === Comparators.NE.label) {
+              value = !!value;
```

Comment:
`!!` - haven't seen this syntax before - is this a way to coerce the value to a boolean?

Reply:
Yup... guess it shouldn't be necessary anymore though, since I've already definitely coerced it to a boolean.  (I had a different way of doing `value` in an earlier revision of this.)

Reply:
Oops, actually that should be just `!value` I think.  Hrm, maybe test coverage isn't good enough.  :)

Reply:
Latest PR should fix this, and has test coverage.